libraries/chain/webassembly/runtimes/eos-vm.cpp

```diff
@@ -129,37 +130,67 @@ class eos_vm_instantiated_module : public wasm_instantiated_module_interface {
          _runtime(runtime),
          _instantiated_module(std::move(mod)) {}
 
+      void do_sync_call(apply_context& context) override {
+         backend_t                                bkend;
+         typename eos_vm_runtime<Impl>::context_t exec_ctx;
+         vm::wasm_allocator                       wasm_alloc;
```

Comment:
It would be preferable to not create and destroy this for every call; it's "expensive" from the standpoint of allocating and freeing pages each time.

Can leave like this for now but,

Have you checked that this is not leaking memory? `wasm_allocator` requires an explicit `free()` call to free its resources  and I don't believe it's called anywhere.

Reply:
> `wasm_allocator` requires an explicit `free()` call to free its resources 

Odd that it `mmap`s in its constructor and doesn't have a destructor to `munmap`.

Reply:
Yeah I do suspect free() should just be moved to be the dtor. Especially since it looks like free() isn't being called anywhere. (I haven't investigated it much though)

Reply:
Thanks. I changed to use a pre-created pool of allocators to avoid expensive allocation for each sync call.

https://github.com/AntelopeIO/spring/pull/1251/commits/48c3e592215a318b7b950b505722a54e53bfbd4d